arm/Microsoft.Automation/automationAccounts/deploy.bicep

```diff
@@ -0,0 +1,254 @@
+})
+param modules array = []
+
+@description('Optional. List of runbooks to be created in the automation account')
+param runbooks array = []
+
+@description('Optional. SAS token validity length. Usage: \'PT8H\' - valid for 8 hours; \'P5D\' - valid for 5 days; \'P1Y\' - valid for 1 year. When not provided, the SAS token will be valid for 8 hours.')
+param sasTokenValidityLength string = 'PT8H'
+
+@description('Optional. List of schedules to be created in the automation account')
+param schedules array = []
+
+@description('Optional. List of jobSchedules to be created in the automation account')
+param jobSchedules array = []
+
+@description('Optional. Configuration Details for private endpoints.')
+param privateEndpoints array = []
+
+@minValue(0)
+@maxValue(365)
+@description('Optional. Specifies the number of days that logs will be kept for; a value of 0 will retain data indefinitely.')
+param diagnosticLogsRetentionInDays int = 365
+
+@description('Optional. Resource identifier of the Diagnostic Storage Account.')
+param diagnosticStorageAccountId string = ''
+
+@description('Optional. Resource identifier of Log Analytics.')
+param workspaceId string = ''
+
+@description('Optional. Resource ID of the event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.')
+param eventHubAuthorizationRuleId string = ''
+
+@description('Optional. Name of the event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category.')
+param eventHubName string = ''
+
+@description('Optional. Switch to lock Automation Account from deletion.')
+param lockForDeletion bool = false
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'')
+param roleAssignments array = []
+
+@description('Optional. Tags of the Automation Account resource.')
+param tags object = {}
+
+@description('Optional. Time used as a basis for e.g. the schedule start date')
+param baseTime string = utcNow('u')
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+var accountSasProperties = {
+  signedServices: 'b'
+  signedPermission: 'r'
+  signedExpiry: dateTimeAdd(baseTime, sasTokenValidityLength)
+  signedResourceTypes: 'o'
+  signedProtocol: 'https'
+}
+var diagnosticsMetrics = [
+  {
+    category: 'AllMetrics'
+    timeGrain: null
+    enabled: true
+    retentionPolicy: {
+      enabled: true
+      days: diagnosticLogsRetentionInDays
+    }
+  }
+]
+var diagnosticsLogs = [
```

Comment:
If a better solution exists I'd vote to use it from now on and tell all other bicep contributors to use it. Please point me to that in case, but also all templates should use the same. Better to align procedures before then after.

Reply:
Agreed. I'm just awaiting @MariusStorhaugs feedback on the matter. The improved approach he implemented can be found [here](https://github.com/Azure/ResourceModules/blob/main/arm/Microsoft.Insights/diagnosticSettings/deploy.bicep), where he essentially enable the user to define the logs on a case-by-case basis with 'all' as the default

Reply:
Same for metrics.

Reply:
Agree, if we just want to use the current set of parameters with the set retention time i would vote to parameterize only the name we want to turn on, then provide our suggestion as default parameters. Would leave the consumer in a place to choose what to send where.

Reply:
Sounds good to me, but I'd vote to do that in a different PR and align also other resource templates

Reply:
Added [PBI 73774](https://unifiedactiontracker.visualstudio.com/Solution%20Engineering/_workitems/edit/73774)

Reply:
alright then